src/core_ocean/shared/mpas_ocn_tendency.F

```diff
@@ -208,175 +208,304 @@ end subroutine ocn_tend_thick!}}}
+                                                 divergence)
+      call mpas_pool_get_array(diagnosticsPool, 'montgomeryPotential', &
+                                                 montgomeryPotential)
+      call mpas_pool_get_array(diagnosticsPool, 'pressure', &
+                                                 pressure)
+      call mpas_pool_get_array(diagnosticsPool, 'vertViscTopOfEdge', &
+                                                 vertViscTopOfEdge)
+      call mpas_pool_get_array(diagnosticsPool, 'density', &
+                                                 density)
+      call mpas_pool_get_array(diagnosticsPool, 'potentialDensity', &
+                                                 potentialDensity)
+      call mpas_pool_get_array(diagnosticsPool, 'tangentialVelocity', &
+                                                 tangentialVelocity)
+      call mpas_pool_get_array(diagnosticsPool, 'surfaceFluxAttenuationCoefficient', &
+                                                 sfcFlxAttCoeff)
+      call mpas_pool_get_array(diagnosticsPool, 'inSituThermalExpansionCoeff', &
+                                                 thermExpCoeff)
+      call mpas_pool_get_array(diagnosticsPool, 'inSituSalineContractionCoeff',  &
+                                                 salineContractCoeff)
+      call mpas_pool_get_array(diagnosticsPool, 'topDrag', &
+                                                 topDrag)
+      call mpas_pool_get_array(diagnosticsPool, 'topDragMagnitude', &
+                                                 topDragMag)
+      call mpas_pool_get_array(diagnosticsPool, 'wettingVelocity', &
+                                                 wettingVelocity)
+      call mpas_pool_get_array(forcingPool, 'windStressZonal', &
+                                             windStressZonal)
+      call mpas_pool_get_array(forcingPool, 'windStressMeridional', &
+                                             windStressMerid)
+
+      call mpas_pool_get_array(tendPool, 'normalVelocity', tendVel)
+
+      call mpas_pool_get_array(forcingPool, 'surfaceStress', &
+                                             sfcStress)
+      call mpas_pool_get_array(forcingPool, 'surfaceStressMagnitude', &
+                                             sfcStressMag)
+
+      !*** Transfer data to device
+      !$acc enter data &
+      !$acc    copyin(tendVel, sfcStress, sfcStressMag, &
+      !$acc    windStressZonal, windStressMerid, topDrag, topDragMag, &
+      !$acc    sfcFlxAttCoeff, ssh, layerThickEdge, normalVelocity, &
+      !$acc    tangentialVelocity, density, potentialDensity, zMid, &
+      !$acc    pressure, layerThickness, circulation, activeTracers, &
+      !$acc    relativeVorticity, kineticEnergyCell, normRelVortEdge, &
+      !$acc    normPlanetVortEdge, montgomeryPotential, &
+      !$acc    vertAleTransportTop, divergence, vertViscTopOfEdge, &
+      !$acc    thermExpCoeff, salineContractCoeff, wettingVelocity)
```

Comment:
Should these OpenACC `data` statements be wrapped in an `#ifdef MPAS_OPENACC`?

Reply:
It's not necessary - with no OpenACC on, these will be ignored. The other ifdefs were needed because they need to disable the openmp directives around the same loops. And sorry about the ifdefs not being in column 1 - it's so much cleaner aligned with the code, so thought I could get away with it.

Reply:
Yeah, I didn't think they needed the preprocessor but wanted to confirm.  Thanks for the clarification.